plugin/storage/cassandra/schema/schema.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package schema
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"regexp"
+	"strconv"
+	"text/template"
+
+	"github.com/jaegertracing/jaeger/pkg/cassandra"
+	"github.com/jaegertracing/jaeger/pkg/cassandra/config"
+)
+
+//go:embed v004-go-tmpl.cql.tmpl
+//go:embed v004-go-tmpl-test.cql.tmpl
+var schemaFile embed.FS
+
+func DefaultSchema() config.Schema {
+	return config.Schema{
+		Keyspace:          "jaeger_v2_test",
+		Datacenter:        "test",
+		TraceTTL:          172800,
+		DependenciesTTL:   0,
+		ReplicationFactor: 1,
+		CasVersion:        4,
+	}
+}
+
+func applyDefaults(cfg *config.Schema) {
+	defaultSchema := DefaultSchema()
+
+	if cfg.Keyspace == "" {
+		cfg.Keyspace = defaultSchema.Keyspace
+	}
+
+	if cfg.Datacenter == "" {
+		cfg.Datacenter = defaultSchema.Datacenter
+	}
+
+	if cfg.TraceTTL == 0 {
+		cfg.TraceTTL = defaultSchema.TraceTTL
+	}
+
+	if cfg.ReplicationFactor == 0 {
+		cfg.ReplicationFactor = defaultSchema.ReplicationFactor
+	}
+
+	if cfg.CasVersion == 0 {
+		cfg.CasVersion = 4
+	}
+}
+
+// Applies defaults for the configs and contructs other optional parameters from it
+func constructCompleteSchema(cfg *config.Schema) error {
+	applyDefaults(cfg)
+
+	cfg.Replication = fmt.Sprintf("{'class': 'NetworkTopologyStrategy', '%s': '%v' }", cfg.Datacenter, cfg.ReplicationFactor)
+
+	if cfg.CompactionWindow != "" {
+		isMatch, err := regexp.MatchString("^[0-9]+[mhd]$", cfg.CompactionWindow)
+		if err != nil {
+			return err
+		}
+
+		if !isMatch {
+			return fmt.Errorf("Invalid compaction window size format. Please use numeric value followed by 'm' for minutes, 'h' for hours, or 'd' for days")
+		}
+
+		cfg.CompactionWindowSize, err = strconv.Atoi(cfg.CompactionWindow[:len(cfg.CompactionWindow)-1])
+		if err != nil {
+			return err
+		}
+
+		cfg.CompactionWindowUnit = cfg.CompactionWindow[len(cfg.CompactionWindow)-1:]
+	} else {
+		traceTTLMinutes := cfg.TraceTTL / 60
+
+		cfg.CompactionWindowSize = (traceTTLMinutes + 30 - 1) / 30
+		cfg.CompactionWindowUnit = "m"
```

Comment:
below you use "MINUTES", is "m" also valid?

Reply:
Yes, m, h and d